src/electron/main.js

```diff
@@ -502,11 +502,47 @@ function registerIPCMainEvents() {
     });
   });
 
-  ipcMain.on(ipcMessages.CANCEL_ROUND, () => {
+  ipcMain.on(ipcMessages.CANCEL_ROUND, event => {
+    let { competition, matchesWithPlayers } = selectedCompetition;
+
     // check if it's a valid state transition (double check if all games are finished?)
+    if (
+      competition.state === COMPETITION_STATE.COMP_ACTIVE_ROUND_READY ||
+      competition.state === COMPETITION_STATE.COMP_ACTIVE_ROUND_ACTIVE
+    ) {
```

Comment:
im Browser etwas schwer zu erkennen. falls alles was du in dem `CANCEL_ROUND` listener machst nur bei comp-active-round-ready oder comp-active-round-active ausgeführt wird könntest du auch einfach so vorgehen:
```
ipcMain.on(....CANCEL_ROUND, event => {
    if ( ... wrong state... ) {
        return;
    }

    // ... do all the work
})
```
Sonst erwarte ich irgendwie dass nach dem if-statement noch was kommt

Reply:
Es gibt zwei Fälle in denen ist es mögich abzubrechen und vier Fälle in denen es nicht möglich ist. Ich fand das einfach zu schreiben. Sonst hat man vier oder Bedinungen.

Reply:
wenn du deine negierst bleibst du bei zwei: `competition.state !== COMP_ACTIVE_ROUND_READY && competition.state !== COMP_ACTIVE_ROUND_ACTIVE` ;)

Reply:
macht aber vom Programm her keinen Unterschied, das stimmt :)